dashboard/app/controllers/reports_controller.rb

```diff
@@ -29,6 +30,28 @@ def header_stats
     render file: 'shared/_user_stats', layout: false, locals: {user: current_user}
   end
 
+  def user_progress
+
+    # TODO: What kind of errors should we be throwing?
+    script_name = params[:script_name]
+    stage_id = params[:stage_id]
+    level_id = params[:level_id]
+    raise ActiveRecord::RecordNotFound unless script_name && level_id
```

Comment:
Followed up offline, this is an unrelated bug in our routing.  ActiveRecord::RecordNotFound is the correct way to signal 404 here.